Runner/Sandbox.cs

```diff
@@ -184,26 +167,19 @@ private void SetAppConfigIfExists()
 
         private void EnumerateAndLoadAssemblies()
         {
-            ScannedAssemblies = Directory.EnumerateFiles(Utils.GetGaugeBinDir(), "*.dll", SearchOption.TopDirectoryOnly)
-                .Select(s =>
-                {
-                    Logger.Debug("Loading assembly from : {0}", s);
-                    return Assembly.Load(AssemblyName.GetAssemblyName(s));
-                })
-                .ToList();
+            foreach (var path in Directory.EnumerateFiles(Utils.GetGaugeBinDir(), "*.dll", SearchOption.TopDirectoryOnly))
+                AssemblyScanner.TryAdd(path);
         }
 
         private void LocateTargetLibAssembly()
         {
-            TargetLibAssembly = ScannedAssemblies.First(assembly => assembly.GetName().Name == GaugeLibAssembleName);
+            TargetLibAssembly = typeof(Step).Assembly;
```

Comment:
I didn't know the reaseon for this. Probably It would have been clearer if the Step type of the TargetLibAssembly would be used instead of the TargetLibAssembly itself.

Btw. I think it could be done in a more elegant way by scanning the gauge_bin folder for the assemblies referencing the lib and using the assembly name to locate the targetlibassembly. Something like (pseudocode):

```
AssemblyName targetLibName = everythingFromGaugeBin.GetReferencedAssemblies()
.Where(a => a.Name == typeof(Step).Assembly.Name).Unique();
TargetLibAssembly = Assembly.TryLoad(targetLibName);
```